ta_services/secure_key_services/include/sks_ta.h

```diff
@@ -14,13 +14,615 @@
 			{ 0xa4, 0x9c, 0xbb, 0xd8, 0x27, 0xae, 0x86, 0xee } }
 
 /*
- * SKS_CMD_PING		Acknowledge TA presence
+ * SKS trusted application may receive requests requesting a number of
+ * parameters that may not suit the GPD 4 parameters directives.
+ * To factorize the use of GPD parameters for SKS services, the current
+ * API uses the 4 GPD invocation parameters as following:
+ *
+ * - Parameter #0 is not used or is used as an input memory reference argument.
+ * The referred buffer hold the command directives arguments (handlers,
+ * ids, ...) expected by the command. When several arguments are expected,
+ * they are serialized with byte alignment in the single control buffer.
+ *
+ * When caller defines parameter #0 as an in/out memory reference, the
+ * parameter #0 is used to feedback caller with a detailed 32bit return
+ * code that allow to return finer error reporting to client than the GPD
+ * TEE API allows a TA to return. For this reason, the invocation commands
+ * below mark parameter #0 as "in(*)-memref" in the command description
+ * comments.
+ *
+ * - Parameter #1 is not used or is an input memory reference. It refers to
+ * the input data provided for the invoked service.
+ *
+ * - Parameter #2 is not used or is an output memory reference. It refers to
+ * the output buffer expected to be filled with the output data requested
+ * by the caller. These data can be a object handle, a ciphered stream, etc.
+ *
+ * - Parameter #3 is not used.
+ */
+
+/*
+ * SKS_CMD_PING		Acknowledge TA presence and return TA versioning info
  *
  * param#0: none
  * param#1: none
- * param#2: none
+ * param#2: none | out-memref : [uint32_t version1]
+ *				[uint32_t version2]
  * param#3: none
  */
 #define SKS_CMD_PING			0x00000000
 
+/*
+ * SKS_CMD_CK_SLOT_LIST - Get the table of the valid slot IDs
+ *
+ * param#0: none
+ * param#1: none
+ * param#2: out-memref : [uint32_t slot_ids]
+ * param#3: none
+ *
+ * The TA instance may represent several PKCS#11 slots and associated tokens.
+ * This command relates the PKCS#11 API function C_GetSlotList and return the
+ * valid IDs recognized by the trsuted application.
+ */
+#define SKS_CMD_CK_SLOT_LIST		0x00000001
+
+/*
+ * SKS_CMD_CK_SLOT_INFO - Get cryptoki structured slot information
+ *
+ * param#0: in(*)-memref : [uint32_t slot_id]
+ * param#1: none
+ * param#2: out-memref : [struct sks_ck_slot_info info]
+ * param#3: none
+ *
+ * The TA instance may represent several PKCS#11 slots and associated tokens.
+ * This command relates the PKCS#11 API function C_GetSlotInfo and return the
+ * information abut the target slot.
+ */
+#define SKS_CMD_CK_SLOT_INFO		0x00000002
+
+struct sks_ck_slot_info {
+	uint8_t slotDescription[64];
```

Comment:
May be replaced by 
#define SKS_SLOT_DESC_SIZE 64
uint8_t slotDescription[SKS_SLOT_DESC_SIZE] just like SKS_TOKEN_LABEL_SIZE

same can be done for manufacturerID

Reply:
ok, done.